### PR TITLE
preparing for new release (with semantic extraction)

### PR DESCRIPTION
**Pull request by anandkp92**
Enables extracting of semantic information from instance annotations. 
More information about how to include annotations can be found [here](https://obc.lbl.gov/specification/cdl.html#semantic-information).

### CODE REVIEW

File: test/reference/json/Buildings/Controls/OBC/CDL/Continuous/PIDWithReset.json [hunk 1]
```diff
@@ -4948,12 +4948,12 @@
...
                                     ],
                                     "textString": "\"PI\"",
-                                    "visible": "([object Object])",
-                                    "lineColor": {
+                                    "textColor": {
                                       "r": 0,
                                       "g": 0,
                                       "b": 0
                                     },
+                                    "visible": "([object Object])",
```
Comment (JayHuLBL) [inline test/reference/json/Buildings/Controls/OBC/CDL/Continuous/PIDWithReset.json:4956]: Should it be changed in different PR?

File: test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Abs.json [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+                      "y": 20
+                    }
+                  ]
+                }
+              }
+            }
+          }
+        }
+      ],
+      "semantics": {},
```
Comment (JayHuLBL) [inline test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Abs.json:371]: should it be empty?

File: test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Abs.json [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+      "within": "Buildings.Controls.OBC.CDL.Continuous",
+      "fullMoFilePath": "/Users/akprakash/Programming/modelica/modelica-buildings/Buildings/Controls/OBC/CDL/Continuous/Abs.mo"
+    }
+  },
+  "requiredReferences": {
+    "connections": {}
```
Comment (JayHuLBL) [inline test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Abs.json:437]: empty object?

File: test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Sources/Constant.json [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+                                {
+                                  "x": -150,
+                                  "y": -150
+                                },
+                                {
+                                  "x": 150,
+                                  "y": -110
+                                }
+                              ],
+                              "textString": ",textString=",
```
Comment (JayHuLBL) [inline test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Sources/Constant.json:233]: should it be a separate issue?

File: test/reference/objects/Buildings/Controls/OBC/CDL/Interfaces/BooleanOutput.json [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+              }
+            }
+          ]
+        }
+      },
+      "within": "Buildings.Controls.OBC.CDL.Interfaces",
+      "fullMoFilePath": "/Users/akprakash/Programming/modelica/modelica-buildings/Buildings/Controls/OBC/CDL/Interfaces/BooleanOutput.mo"
+    }
+  },
+  "requiredReferences": {}
```
Comment (JayHuLBL) [inline test/reference/objects/Buildings/Controls/OBC/CDL/Interfaces/BooleanOutput.json:232]: empty object?

**Review by JayHuLBL — COMMENTED** [review]
@anandkp92 I dropped some inline comments. Not sure the issues are related to this PR, or they should be addressed in separated PRs in future.

File: test/reference/json/Buildings/Controls/OBC/CDL/Continuous/PIDWithReset.json (hunk 1 above)
Comment (anandkp92) [reply test/reference/json/Buildings/Controls/OBC/CDL/Continuous/PIDWithReset.json:4956]: created a new issue to address this (https://github.com/lbl-srg/modelica-json/issues/212)

File: test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Abs.json (hunk 2 above)
Comment (anandkp92) [reply test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Abs.json:371]: yes. some parts of the code assume it exists (even if its empty). Can discuss and change in the future if needed

File: test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Abs.json (hunk 3 above)
Comment (anandkp92) [reply test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Abs.json:437]: yes. some parts of the code assume it exists (even if its empty). Can discuss and change in the future if needed

File: test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Sources/Constant.json (hunk 4 above)
Comment (anandkp92) [reply test/reference/objects/Buildings/Controls/OBC/CDL/Continuous/Sources/Constant.json:233]: Yes. created it here: https://github.com/lbl-srg/modelica-json/issues/213

File: test/reference/objects/Buildings/Controls/OBC/CDL/Interfaces/BooleanOutput.json (hunk 5 above)
Comment (anandkp92) [reply test/reference/objects/Buildings/Controls/OBC/CDL/Interfaces/BooleanOutput.json:232]: yes. some parts of the code assume it exists (even if its empty). Can discuss and change in the future if needed
